homeassistant/components/light/wemo.py

```diff
@@ -140,3 +148,89 @@ def turn_off(self, **kwargs):
     def update(self):
         """Synchronize state with bridge."""
         self.update_lights(no_throttle=True)
+
+
+class WemoDimmer(Light):
+    """Representation of a WeMo dimmer."""
+
+    def __init__(self, device):
+        """Initialize the WeMo dimmer."""
+        self.wemo = device
+        self._brightness = None
+        self._state = None
+        # look up model name once as it incurs network traffic
+        self._model_name = self.wemo.model_name
+
+        wemo = get_component('wemo')
+        wemo.SUBSCRIPTION_REGISTRY.register(self.wemo)
+        wemo.SUBSCRIPTION_REGISTRY.on(self.wemo, None, self._update_callback)
+
+    def _update_callback(self, _device, _type, _params):
+        """Update the state by the Wemo device."""
+        _LOGGER.info("Subscription update for  %s", _device)
+        updated = self.wemo.subscription_update(_type, _params)
+        self._update(force_update=(not updated))
+
+        if not hasattr(self, 'hass'):
+            return
+        self.schedule_update_ha_state()
+
+    @property
+    def unique_id(self):
+        """Return the ID of this WeMo dimmer."""
+        return "{}.{}".format(self.__class__, self.wemo.serialnumber)
+
+    @property
+    def name(self):
+        """Return the name of the dimmer if any."""
+        return self.wemo.name
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 1 and 100."""
+        wemobrightness = int(self.wemo.get_brightness())
+        return int((wemobrightness * 255) / 100)
+
+    @property
+    def is_on(self):
+        """Return true if dimmer is on. Standby is on."""
+        return self._state
+
+    def update(self):
```

Comment:
Do you still need to have polling active, since you register a callback for updates in pywemo? Maybe you should have the property `should_poll` return `False`?

Reply:
Added this in my recent update to the code

Reply:
Now you can remove this method, right?